### PR TITLE
refactor default reporter, add report formatter, add tests

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**

### CODE REVIEW

**Comment by HlebMaliborski** [comment]
Let's add documentation to existing functionality. We can do it within PR's if we touch some classes.

**Comment by alexander-kulikovskii** [comment]
> Let's add documentation to existing functionality. We can do it within PR's if we touch some classes.

Let's merge as is. Documentation will be in separate PR

File: lib/src/main/java/fi/epicbot/toster/model/ReportConfig.kt [hunk 1]
```diff
@@ -2,8 +2,15 @@ package fi.epicbot.toster.model
 
 import fi.epicbot.toster.report.DefaultReporter
 import fi.epicbot.toster.report.Reporter
+import fi.epicbot.toster.report.formatter.JsonFormatter
 
 class ReportConfig(
     var enable: Boolean = true,
-    var reporters: MutableList<Reporter> = mutableListOf(DefaultReporter()),
+    var reporters: MutableList<Reporter> = mutableListOf(
```
Comment (HlebMaliborski) [inline lib/src/main/java/fi/epicbot/toster/model/ReportConfig.kt:9]: Maybe we should keep posibility to set Reporter in config, not from outside?

Comment (alexander-kulikovskii) [reply lib/src/main/java/fi/epicbot/toster/model/ReportConfig.kt:9]: leave such reporters only as user custom reporters and add default only when use them all and pass it as extra parameter?

File: lib/src/main/java/fi/epicbot/toster/report/formatter/JsonFormatter.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package fi.epicbot.toster.report.formatter
+
+import fi.epicbot.toster.report.model.ReportOutput
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+class JsonFormatter(private val prettyPrintJson: Boolean) : ReportFormatter {
+
+    override fun format(reportOutput: ReportOutput): String {
```
Comment (HlebMaliborski) [inline lib/src/main/java/fi/epicbot/toster/report/formatter/JsonFormatter.kt:9]: Is it neccessary to have this parameter so dynamic? Maube pass it to constructor?

Comment (alexander-kulikovskii) [reply lib/src/main/java/fi/epicbot/toster/report/formatter/JsonFormatter.kt:9]: I can't see any difference. with one formatter you can format different outputs

**Review by HlebMaliborski — APPROVED** [review]

**Review by HlebMaliborski — APPROVED** [review]

**Pull request merged**